Add mask and unmask option to --security-opt (#8408)
containers/podman: docs/source/markdown/podman-run.1.md

- **apparmor=unconfined** : Turn off apparmor confinement for the container
- **apparmor**=_your-profile_ : Set the apparmor confinement profile for the container
- **mask**=_/path/1:/path/2_: The paths to mask separated by a colon. A masked path

>   cannot be accessed inisde the container.

Reviewer: ```suggestion
  cannot be accessed inside the container.
```

Reviewer: and sorry if that was my fat fingers!

Author: Fixed, thanks! I should enable spell checker :D

- **seccomp**=_profile.json_:  Allowed syscall list seccomp JSON file to be used as a seccomp filter
- **proc-opts**=_OPTIONS_ : Comma separated list of options to use for the /proc mount.  More details
  for the possible mount options are specified at **proc(5)** man page.
- **unmask**=_ALL_ or _/path/1:/path/2_: Paths to unmask separated by a colon. If set to **ALL**, it will

>   unmask all the paths that are masked by default.

Reviewer 2: Can we mention which paths are marked by default? May be worth adding a comment in the code as well to keep the man page and the `[]string` in sync.

Reviewer 3: Can you also explain what a masked path is? Users may not be familiar?

Author: Fixed

- **apparmor=unconfined** : Turn off apparmor confinement for the container
- **apparmor**=_your-profile_ : Set the apparmor confinement profile for the container

> - **mask**=_/path/1:/path/2_: The paths to mask separated by a colon. A masked path

Reviewer: can you move this below the labels?

Author: Fixed

- **seccomp**=_profile.json_: Allowed syscall list seccomp JSON file to be used as a seccomp filter
- **proc-opts**=_OPTIONS_ : Comma separated list of options to use for the /proc mount. More details
  for the possible mount options are specified at **proc(5)** man page.
- **unmask**=_ALL_ or _/path/1:/path/2_: Paths to unmask separated by a colon. If set to **ALL**, it will
  unmask all the paths that are masked by default.

>   The default masked paths are **/proc/acpi,/proc/kcore,/proc/keys,/proc/latency_stats,/proc/timer_list,/proc/timer_stats,/proc/sched_debug,/proc/scsi,/sys/firmware,/sys/fs/selinux.**

Reviewer: moved timer_stats and added space.
```suggestion
  The default masked paths are **/proc/acpi, /proc/kcore, /proc/keys, /proc/latency_stats, /proc/timer_list, /proc/sched_debug, /proc/scsi, /proc/timer_stats, /sys/firmware, and /sys/fs/selinux.**
```

Author: Fixed

Note you would have to write policy defining a **svirt_apache_t** type.

> If you want to additionaly mask specific paths in the container, you can specify

Reviewer 2: ```suggestion
If you want to additionally mask specific paths in the container, you can specify
```

Reviewer: No "you" or similar pronouns in man pages, please.  Yet something else we're not consistent with (and a side project I need to do to clean).  Perhaps:
```suggestion
To mask additional specific paths in the container,  specify
```

Author: Fixed

the paths separated by a colon using the **mask** option with the **--security-opt**
flag.

```
$ podman run --security-opt mask=/foo/bar:/second/path fedora bash
```

> If you want to unmask all the paths that are masked by default, you can set the **unmask**

Reviewer: ditto "you"

Author: Fixed
